wordpress-mobile/WordPress-Android
# Aztec - add VideoPress support

This PR adds support for VideoPress shortcodes and show the correct thumbnail for videos.

Both editing a post that contains a VP video, or uploading a new video to wpcom sites, should show the VideoPress shortcode correctly in HTML mode, and the preview thumb (a.k.a posterURL) when the editor is in Visual mode.

cc @mzorz since this PR does interact with the uploading service of WP. Also ping him because I see the attribute `wp-data-id` kept left in the shortcode after the uploading. Put a breakpoint in `onMediaUploadSucceeded` and see that the attribute above is not removed on videos. I think we should remove all of those used for pictures on videos.

Note that this PR https://github.com/wordpress-mobile/AztecEditor-Android/pull/575 should be merged first, and the hash updated here. 

** To test**

*Test 1*
- Start a new post
- Upload a video to a wpcom blog
- Wait till the end of the video, switch to html mode, and check that the correct shortcode is there
- Switch back to visual, the thumb URL should be loaded now
(*) Note that the thumb URL returned from the videopress service could be different from the one we have locally. A fix for this will be adddressed in a different PR. (We're also having changing in size here, but there is already a ticket for this other issue).

*Test 2*
- Open a post with a VP video in it
- The visual editor should show the correct thumb
- Switch to HTML mode. The VP shortcode should be fine there
- Switch back to visual, the thumb URL should be loaded again
